src/eda_functions.py

```diff
@@ -0,0 +1,179 @@
+
+    Args:
+        xgb_cm (sklearn.metrics.confusion_matrix): The confusion matrix to be plotted.
+        labels (list): List of the class label names.
+    """
+
+    # setup the heatmap of the confusion matrix
+    ax = sns.heatmap(xgb_cm, annot=True, cmap=cmap, fmt='g')
+    ax.set_title('Confusion Matrix with labels\n');
+    ax.set_xlabel('\nPredicted Values')
+    ax.set_ylabel('Actual Values ');
+
+    # set the ticket labels
+    ax.xaxis.set_ticklabels(labels)
+    ax.yaxis.set_ticklabels(labels)
+    plt.yticks(rotation=0) 
+    plt.xticks(rotation=90, horizontalalignment='center') 
+
+    # display the confusion matrix
+    plt.show()
+
+
+def get_label_accuracies(xgb_cm, labels):
+    # create empty lists to catch the results
+    accuracies = []
+    often_confused_with = []
+    confusion_percent = []
+
+    for idx, label in enumerate(labels):
+      # calculate the accuracy for the current label
+      highest_value = np.diagonal(xgb_cm)[idx]
+      sum_of_labels = np.sum(xgb_cm[idx])
+      label_accuracy = round(highest_value/sum_of_labels, 2)
+      accuracies.append(label_accuracy)
+
+      # identify the label that the current one is most often confused with
+      second_highest_value = np.partition(xgb_cm[idx].flatten(), -2)[-2]
+      second_highest_value_idx = (np.where(xgb_cm[idx] == second_highest_value))
+      often_confused_with.append(labels[int(second_highest_value_idx[0])])
+
+      # get the confusion in percentage
+      confusion = round(second_highest_value/highest_value*100, 2)
+      confusion_percent.append(confusion)
+
+    # create a data frame using a dictionary
+    data = {
+        'crop_type':labels, 
+        'accuracy':accuracies,
+        'often_confused_with':often_confused_with,
+        'confusion_percent':confusion_percent
+        }
+    acc_df = pd.DataFrame(data)
+
+    # sort the data frame in order decreasing feature importance
+    acc_df.sort_values(by=['accuracy'], ascending=False, inplace=True)
+
+    return acc_df
+
+
+def plot_label_accuracy(df):
+    # plot Searborn bar chart
+    ax = sns.barplot(data=df, x='crop_type', y='accuracy', color=cmap[4])
+
+    # do the annotation on each bar
+    for bar in ax.patches:
+        ax.annotate(format(bar.get_height(), '.2f'), 
+                      (bar.get_x() + bar.get_width() / 2., bar.get_height()), 
+                      ha = 'center', va = 'center', 
+                      xytext = (0, 9), 
+                      textcoords = 'offset points')
+
+    # set chart labels
+    ax.set_title('Accuracy of the different crop types\n')
+    ax.set_xlabel('')
+    ax.set_ylabel('Accuracy')
+
+    # set the ticket labels
+    plt.yticks(rotation=0) 
+    plt.ylim(0, 1.1)
+    plt.xticks(rotation=90, horizontalalignment='center')
+
+    # display the bar chart of the label accuracies
+    plt.show()
+
+
+def plot_feature_importance(importance, feature_names, model_name, num_features=None):
```

Comment:
could you add what kind of data the importance should be:

SO like: importance: list

Reply:
Its added. :)